ui/src/app/utils/subscribable.tsx

```diff
@@ -0,0 +1,43 @@
+import * as fp from 'lodash/fp';
+
+export interface Subscribable<T> {
+  subscribe: (Function) => { unsubscribe: () => void};
+  next: (newValue: T, oldValue: T) => void;
+}
+
+export interface Atom<T> {
+  get: () => T;
+  set: (value: T) => void;
+  subscribe: Subscribable<T>['subscribe'];
+  reset: () => void;
+}
+
+export function subscribable<T>(): Subscribable<T> {
+  let subscribers = [];
+  return {
+    subscribe: fn => {
```

Comment:
Is this based on something else? There's some interesting design considerations here. Is this tailored for a particular constrained use case - or if not, do you have rationale on the following?

- Whether or not to immediately deliver the most recent element to new subscribers. For something like an event system, this wouldn't make sense - but for a store, it can make the initial connection boilerplate a bit more homogeneous (one subscribe() path, rather than a get + subscribe).
- Synchronous vs asynchronous handling on `next()`, i.e. current approach vs invoking subscribers in a `setTimeout(, 0)` to pop this out of the stack. This provides some incapsulation between the subscribers, and avoids dependency on synchronous `next()` behaviors.